alfalfa_worker/lib/run.py

```diff
@@ -0,0 +1,91 @@
+import glob
+import os
+from datetime import datetime
+from enum import auto
+from typing import List
+from uuid import uuid4
+
+import pytz
+from dateutil import parser
+
+from alfalfa_worker.lib.auto_name_enum import AutoName
+from alfalfa_worker.lib.point import Point
+from alfalfa_worker.lib.sim_type import SimType
+
+
+class RunStatus(AutoName):
+    CREATED = auto()
+    PREPROCESSING = auto()
+    READY = auto()
+    STARTING = auto()
+    STARTED = auto()
+    RUNNING = auto()
+    STOPPING = auto()
+    COMPLETE = auto()
+    ERROR = auto()
+
+
+class Run:
+    # A lot of stuff here is done to store in the db. It is messy. If we are sticking with mongo db or switching to something else it would be prettied.
+
+    def __init__(self, dir=None, model=None, _id=None, job_history=[], sim_type=SimType.OPENSTUDIO, status=RunStatus.CREATED, created=None, modified=None, sim_time=None):
+        self.dir = dir
+        self.model = model
+        self.id = _id if _id is not None else str(uuid4())
+        self._job_history = job_history
+        self._status = status if status.__class__ == RunStatus else RunStatus(status)
+        self.sim_type = sim_type if sim_type.__class__ == SimType else SimType(sim_type)
+        if created is None:
+            created = datetime.now(tz=pytz.UTC)
+        self.created = created if created.__class__ == datetime else parser.parse(created)
+        if modified is None:
+            modified = datetime.now(tz=pytz.UTC)
+        self.modified = modified if modified.__class__ == datetime else parser.parse(modified)
+        self.sim_time = sim_time
+        self.points: List[Point] = []
+
+    def join(self, *args):
```

Comment:
we could move this to pathlib.Path...

Reply:
as in, the run would expose a path object? so people would do `run.dir / 'my.osw'`?

Reply:
yeah, that is what I'm thinking...